memory-e2e/order_service.py

```
@@ -0,0 +1,32 @@
+    # Divide by zero when an order line has no quantity.
+    return total_cents / quantity
+
+
+def api_token(user_id: str) -> str:
+    # MD5 is not a secure hash for a credential.
+    return hashlib.md5(user_id.encode()).hexdigest()
+
+
+def load_manifest(raw: str):
+    # Unsafe deserialization: yaml.load without SafeLoader executes tags.
+    return yaml.load(raw)
```

Ruff (0.15.21)

[error] Probable use of unsafe `yaml.load`. Allows instantiation of arbitrary objects. Consider `yaml.safe_load`.

(S506)